android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/SpeechModule.java

```diff
@@ -0,0 +1,142 @@
+                  .emit("Exponent.speakingStarted", idToMap(utteranceId));
+            }
+
+            @Override
+            public void onDone(String utteranceId) {
+              getReactApplicationContext()
+                  .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                  .emit("Exponent.speakingDone", idToMap(utteranceId));
+            }
+
+            @Override
+            public void onStop(String utteranceId, boolean interrupted) {
+              getReactApplicationContext()
+                  .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                  .emit("Exponent.speakingStopped", idToMap(utteranceId));
+            }
+
+            @Override
+            public void onError(String utteranceId) {
+              getReactApplicationContext()
+                  .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                  .emit("Exponent.speakingError", idToMap(utteranceId));
+            }
+          });
+        }
+      }
+    });
+  }
+
+  @Override
+  public String getName() {
+    return "ExponentSpeech";
+  }
+
+  @ReactMethod
+  public void speak(final String id, final String text, final ReadableMap options) {
+    if (mTtsReady) {
+      if (options.hasKey("language")) {
+        Locale locale = new Locale(options.getString("language"));
+        int languageAvailable = mTextToSpeech.isLanguageAvailable(locale);
+        if (languageAvailable != TextToSpeech.LANG_MISSING_DATA &&
+            languageAvailable != TextToSpeech.LANG_NOT_SUPPORTED) {
+          mTextToSpeech.setLanguage(locale);
+        } else {
+          mTextToSpeech.setLanguage(Locale.getDefault());
+        }
+      } else {
+        mTextToSpeech.setLanguage(Locale.getDefault());
+      }
+      if (options.hasKey("pitch")) {
+        mTextToSpeech.setPitch((float) options.getDouble("pitch"));
+      }
+      if (options.hasKey("rate")) {
+        mTextToSpeech.setSpeechRate((float) options.getDouble("rate"));
+      }
+
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+        mTextToSpeech.speak(
+            text,
+            TextToSpeech.QUEUE_ADD,
+            null,
+            id
+        );
+      } else {
+        HashMap<String, String> params = new HashMap<>();
+        params.put(TextToSpeech.Engine.KEY_PARAM_UTTERANCE_ID, id);
+        mTextToSpeech.speak(
+            text,
+            TextToSpeech.QUEUE_ADD,
+            params
+        );
+      }
+    }
+  }
+
+  @ReactMethod
+  public void isSpeaking(Promise promise) {
+    promise.resolve(mTtsReady && mTextToSpeech.isSpeaking());
+  }
+
+  @ReactMethod
+  public void stop() {
+    if (mTtsReady) {
+      mTextToSpeech.stop();
+    }
+  }
+
+  @Override
+  public void onHostResume() { }
+
+  @Override
+  public void onHostPause() { }
+
+  @Override
+  public void onHostDestroy() {
+    mTextToSpeech.shutdown();
```

Comment:
what does this do? is there a reason we don't need to call this in onHostPause instead?